Source/Particles/MultiParticleContainer.cpp

```diff
@@ -991,6 +1020,114 @@ MultiParticleContainer::BreitWheelerGenerateTable ()
+    auto dVdt = geom.CellSize(0) * geom.CellSize(1) * m_qed_schwinger_y_size
+               * warpx.getdt(level_0);
+#elif (AMREX_SPACEDIM == 3)
+    auto dVdt = geom.CellSize(0) * geom.CellSize(1) * geom.CellSize(2)
+               * warpx.getdt(level_0);
+#endif
+
+    auto& pc_product_ele =
+            allcontainers[m_qed_schwinger_ele_product];
+    auto& pc_product_pos =
+            allcontainers[m_qed_schwinger_pos_product];
+
+    const MultiFab & Ex = warpx.getEfield(level_0,0);
+    const MultiFab & Ey = warpx.getEfield(level_0,1);
+    const MultiFab & Ez = warpx.getEfield(level_0,2);
+    const MultiFab & Bx = warpx.getBfield(level_0,0);
+    const MultiFab & By = warpx.getBfield(level_0,1);
+    const MultiFab & Bz = warpx.getBfield(level_0,2);
+
+    MFItInfo info;
+    if (TilingIfNotGPU()) {
+        info.EnableTiling();
+    }
+#ifdef _OPENMP
+    info.SetDynamic(WarpX::do_dynamic_scheduling);
+#pragma omp parallel
+#endif
+
+    for (MFIter mfi(Ex, info); mfi.isValid(); ++mfi )
+    {
+        // Make the box cell centered to avoid creating particles twice on the tile edges
+        const Box& box = enclosedCells(mfi.tilebox());
+
+        const auto& arrEx = Ex[mfi].array();
+        const auto& arrEy = Ey[mfi].array();
+        const auto& arrEz = Ez[mfi].array();
+        const auto& arrBx = Bx[mfi].array();
+        const auto& arrBy = By[mfi].array();
+        const auto& arrBz = Bz[mfi].array();
+
+        const Array4<const amrex::Real> array_EMFAB [] = {arrEx,arrEy,arrEz,
+                                           arrBx,arrBy,arrBz};
+
+        pc_product_ele->defineAllParticleTiles();
+        pc_product_pos->defineAllParticleTiles();
+
+        auto& dst_ele_tile = pc_product_ele->ParticlesAt(level_0, mfi);
+        auto& dst_pos_tile = pc_product_pos->ParticlesAt(level_0, mfi);
+
+        const auto np_ele_dst = dst_ele_tile.numParticles();
+        const auto np_pos_dst = dst_pos_tile.numParticles();
+
+        const auto Filter  = SchwingerFilterFunc{
+                              m_qed_schwinger_threshold_poisson_gaussian,dVdt};
+
+        const SmartCreateFactory create_factory_ele(*pc_product_ele);
+        const SmartCreateFactory create_factory_pos(*pc_product_pos);
+        const auto CreateEle = create_factory_ele.getSmartCreate();
+        const auto CreatePos = create_factory_pos.getSmartCreate();
+
+        const auto Transform = SchwingerTransformFunc{m_qed_schwinger_y_size,
+                            ParticleStringNames::to_index.find("w")->second};
+
+        const auto num_added = filterCreateTransformFromFAB<1>( dst_ele_tile,
+                              dst_pos_tile, box, array_EMFAB, np_ele_dst,
+                               np_pos_dst,Filter, CreateEle, CreatePos,
+                                Transform);
+
+        setNewParticleIDs(dst_ele_tile, np_ele_dst, num_added);
+        setNewParticleIDs(dst_pos_tile, np_pos_dst, num_added);
+
+        WARPX_PROFILE_VAR_STOP(blp_schwinger);
```

Comment:
Well I defined the variable on purpose so that Schwinger does not pop up on the parser output if Schwinger process is turned off (I figured it might be slightly confusing for a user). Do you think I should remove it?

In any case you're right, the `STOP` is in the for loop. :-/ I'll fix that if we decide to keep the variable.

Reply:
If you put `WARPX_PROFILE` after the conditional return `if (!m_do_qed_schwinger) return;` (i.e., not like my suggestion) does the string still pop up in the Tiny Profiler when the Schwinger process is off?

Even if it does, I think it's nicer to use `WARPX_PROFILE("...")` as this is precisely meant for what we're doing here (profile a whole function). If the behavior is unexpected, we could consider fixing `WARPX_PROFILE`.

Reply:
Ok thanks, I've committed your solution and removed the variable.

To answer your question: yes Schwinger did not appear in the Tiny Profiler when `WARPX_PROFILE` was after the `return` statement and the Schwinger process was turned off. Now (with the newest commit), Schwinger appears in the profiler even when the process is turned off.